### ISSUE TITLE
Splatoon 2 crashes when trying to open souvenir screen/going to multiplayer lobby/changing octoling body

### ISSUE THREAD

**Issue by AntoSkate** [yuzu-emu/yuzu#5307]
Yuzu version 499
Windows 10 pro x64 20H2
GPU: Nvidia RTX 2060 (driver 461.09)

The crashes happened using OpenGL, i'll try with Vulkan later

There are various crashes in splatoon 2, one when entering the lobby, another one when opening a souvenir in octo expansion, and another when changing octoling body in coto expansion.

Lobby log:

[yuzu_log.txt](https://github.com/yuzu-emu/yuzu/files/5783480/yuzu_log.txt)

Souvenir log:

[yuzu_log.txt](https://github.com/yuzu-emu/yuzu/files/5783483/yuzu_log.txt)

Changing octoling body:

[yuzu_log.txt](https://github.com/yuzu-emu/yuzu/files/5783485/yuzu_log.txt)

**Comment by AntoSkate** [comment]
Fixed by: #6585

**Comment by BSoDGamingYT** [comment]
Reopened since this change is not in master

**Comment by Lt-knb** [comment]
Entering the lobby still leads to a crash, at least for me. I tested Early Access 1866

**Comment by Lt-knb** [comment]
All three crashes still happen. Tested EA1887

**Comment by AntoSkate** [comment]
> 
> 
> All three crashes still happen. Tested EA1887

I tried with the latest pr artifact and it works(at least on vulkan, i haven't tried with ogl)
![immagine](https://user-images.githubusercontent.com/36473846/126081065-17e04b58-f654-4082-ba27-7822d573c5f6.png)

**Comment by Lt-knb** [comment]
Tested Mainline 691 and Early Access 1887 and the crashes still occur. My specs are:
Ryzen 7 2700
Gtx 1070
Ram 16 GB
Windows 10 1909
Nvidia Driver Game Ready 457.30

**Comment by Lt-knb** [comment]
Please, either enter the lobby or open a souvenir in Mainline 691 with OpenGL. Post the resulting log if it does _not_ crash.
That would help

**Comment by Lt-knb** [comment]
Finally found out what is actually going on:
This issue is fixed by enabling the Garbage Collector on Vulkan.
OpenGl still crashes

**Comment by v1993** [comment]
> OpenGl still crashes

Even if you enable garbage collection there? Because it works for both backends.

**Comment by Lt-knb** [comment]
Just tested EA1908 and can confirm enabling the Reaper makes no difference on OpenGl. All crashes still occur

**Comment by AntoSkate** [comment]
> 
> 
> > OpenGl still crashes
> 
> Even if you enable garbage collection there? Because it works for both backends.

I tried with self built ea and opengl still crashes even with garbage collector

**Comment by v1993** [comment]
Since hades was said to be unrelated, should this be reopened?

**Comment by Lt-knb** [comment]
Just tested EA1915 and of course it still crashes.
Please reopen

**Comment by AntoSkate** [comment]
> 
> 
> Since hades was said to be unrelated, should this be reopened?

yeah i think it should

**Comment by v1993** [comment]
@AntoSkate Then do it! You should have "reopen" button available.

**Comment by AntoSkate** [comment]
> 
> 
> @AntoSkate Then do it! You should have "reopen" button available.

I don't see it, i only have "Edit" and "New issue" buttons

**Comment by v1993** [comment]
Ah, sorry then. It's probably unavailable because it was automatically closed, so we'll have to ask @BSoDGamingYT to do this for us.

**Comment by Lt-knb** [comment]
After #6897 the issue also affects Vulkan (garbage collection no longer fixes it)

**Comment by Rayxxon** [comment]
Tested EA2390 ,running in opengl and it  crashes.
E3 1231V3  DDR3 16G gtx 1070

**Comment by Lt-knb** [comment]
The lobby works, at least on Vulkan.
Mem cakes still crash

**Comment by Lt-knb** [comment]
Just tested and it works now, all 3 of them (both OpenGL and Vulkan), but...
Mem cakes have the wrong shape now (at random). Only on Vulkan though.
Pretty weird regression

**Comment by RexSonic** [comment]
I still have this issue on the newest release

**Comment by Lt-knb** [comment]
Aaand I confirm it crashes again.
All 3 of them...
Oof
